agent/src/agentclient.rs

```diff
@@ -245,20 +244,28 @@ impl<T: APIServerClient> BrupopAgent<T> {
 
             let bottlerocket_node_status = bottlerocket_node
                 .status
-                .context(error::MissingBottlerocketNodeStatus)?;
+                .context(agentclient_error::MissingBottlerocketNodeStatus)?;
             let bottlerocket_node_spec = bottlerocket_node.spec;
 
             // Determine if the spec on the system's custom resource demands the node take action. If so, begin taking that action.
             if bottlerocket_node_spec.state != bottlerocket_node_status.current_state {
-                log::info!("Detected drift between spec state and current state. Requesting node to take action: {:?}.", &bottlerocket_node_spec.state);
+                event!(
+                    Level::INFO,
+                    brn_name = ?bottlerocket_node.metadata.name,
+                    action = ?bottlerocket_node_spec.state,
+                    "Detected drift between spec state and current state. Requesting node to take action"
+                );
 
                 match bottlerocket_node_spec.state {
                     BottlerocketNodeState::Idle => {
-                        log::info!("Ready to finish monitoring and start update process")
+                        event!(
+                            Level::INFO,
+                            "Ready to finish monitoring and start update process"
+                        );
                     }
                     BottlerocketNodeState::StagedUpdate => {
-                        log::info!("Preparing update");
-                        prepare().await.context(error::UpdateActions {
+                        event!(Level::INFO, "Preparing update");
+                        prepare().await.context(agentclient_error::UpdateActions {
```

Comment:
Maybe worth adding an `#[instrument]` to `run()` so that these error cases are logged?